code generators/de.fhdo.lemma.model_processing.code_generation.container_base/src/main/java/de/fhdo/lemma/model_processing/code_generation/container_base/file/docker/OpendDockerComposeFile.xtend

```diff
@@ -0,0 +1,242 @@
+    List<String> dockerComposeNetworkParts
+
+
+    new() {
+        dockerComposeServiceParts = new HashMap
+        dockerComposeNetworkParts = new LinkedList
+    }
+
+    /*
+     * Get or create instance of the [OpendDockerComposeFile] class.
+     */
+    static def OpendDockerComposeFile getInstance() {
+        if (OpendDockerComposeFile.instance === null)
+            OpendDockerComposeFile.instance = new OpendDockerComposeFile
+
+        return OpendDockerComposeFile.instance
+    }
+
+    /*
+     * This method initializes the [OpendDockerComposeFile] class. Therefor the taget folder for the
+     * docker compose file is set. Additionally, an existing docker compose file is loaded from the
+     * file system into the [DockerComposeFile] date structure.
+     */
+    def init(String targetFolder) {
+        dockerComposePath = targetFolder + File.separator + "docker-compose.yml"
+    }
+
+    /*
+     * Load a existing docker compose file form the file system. If there is no docker compose file
+     * present a default network configuration is created, because LEMMA's operation modeling
+     * language does in its actual version not support the handling of network configurations.
+     */
+    def openExistingDockerComposeFile() {
+        if (new File(dockerComposePath).exists && new File(dockerComposePath).length > 0) {
+            loadDockerComposePartsFromFile
+        } else {
+            createDefaultNetwork
+        }
+    }
+
+    /*
+     * Add or replace a docker compose part to the docker compose file based on the name of the
+     * operation node.The configuration of the docker compose part is created by an intermediate
+     * deployment aspect.
+     */
+    def addOrReplaceDockerComposePart(String name, String dockerComposeServicePart) {
+        this.addOrReplaceToMap(name,
+            Util::adjustIndentations('  ', dockerComposeServicePart) + '\n'
+        )
+    }
+
+    /*
+     * Add or replace a docker compose part to the docker compose file based on the name of the
+     * operation node. The configuration of the docker compose part is created by the docker compose
+     * code generation module itself.
+     *
+     * Note: the docker compose part is done separately for intermediate container and intermediate
+     * infrastructure nodes.
+     */
+    def addOrReplaceDockerComposePart(IntermediateOperationNode node) {
+        if (IntermediateInfrastructureNode.isInstance(node)) {
+            addOrReplaceToMap(node.name,
+                getServiceFromInfrastructureNode(node as IntermediateInfrastructureNode) + "\n"
+            )
+        } else {
+            addOrReplaceToMap(node.name,
+                getServiceFromContainer(node as IntermediateContainer) +"\n"
+            )
+        }
+    }
+
+    /*
+     * Create a default network for the docker compose file. This is needed because LEMMA's
+     * operation modeling language does not support network configurations at the moment.
```

Comment:
Would it make sense to have aspects for this issue?

Reply:
I don't think aspects can fix this problem because the network is a configuration that is independent of the components of the Docker-Compose file.

However, the configuration of the operation nodes for the corresponding network can be realized by aspects, but this does not help much since the basic network configuration cannot be modeled.

Reply:
Yeah, exactly your last statement is my actual point: Why can't we influence the peculiarity of a basic network (and maybe additional networks) by means of aspects? Something like
```
@Network(basic=true, option1=foo, optionX=bar...)

Reply:
I will create an issue for this specific topic and fix it in a separate pull request.

Reply:
Excellent. Please create the issue and mark this conversation as resolved.